utils/sysfs/sysfs.go

```diff
@@ -335,25 +352,205 @@ func (fs *realSysFs) GetSystemUUID() (string, error) {
 	}
 }
 
-func (fs *realSysFs) IsCPUOnline(dir string) bool {
-	cpuPath := fmt.Sprintf("%s/online", dir)
-	content, err := ioutil.ReadFile(cpuPath)
+func (fs *realSysFs) IsCPUOnline(cpuPath string) bool {
+	return isCPUOnline(cpuPath)
+}
+
+// isCPUOnline verifies if a CPU in online. Takes path to /sys/devices/system/cpu/cpu*/ directory as a parameter.
+func isCPUOnline(cpuPath string) bool {
+	if isx86() && isZeroCPU(cpuPath) {
+		// It *is* possible to enable CPU hotplug for CPU0 but it requires setting a boot flag.
+		// It is assumed it's a rare situation.
+		return true
+	}
+
+	const isOnline = 49
```

Comment:
just for my own curiosity, why is 49? is is documented in kernel docs somewhere?

Reply:
49 is nothing more than 1, but as a byte rather than a string.

Reply:
got it :) might be more clear to define it maybe characters are or `const isOnline := byte('1')`

Reply:
Fixed.